frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -246,6 +269,13 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
     this.decisionPoints.controls.forEach((_, index) => {
       this.manageSiteAndTargetControls(index);
     });
+
+    // Store current experiment name from Overview step for Mooclet configuration
+    this.subscriptionHandler.add(
+      this.experimentDesignStepperService.currentExperimentName$.subscribe((name) => {
```

Comment:
this should be subscribed directly in the template with `currentExperimentName$ | async`.